File: frontend/src/api/models/ResidentialOwnerUserCreateRequestDTO.ts
import type { Account } from './Account.ts'
import type { ResidentialUnit } from './ResidentialUnit.ts'

export const residentialOwnerUserCreateRequestDTODocumentTypeEnum = {
  CPF: 'CPF',
  CNPJ: 'CNPJ',
} as const

export type ResidentialOwnerUserCreateRequestDTODocumentTypeEnum =
  (typeof residentialOwnerUserCreateRequestDTODocumentTypeEnum)[keyof typeof residentialOwnerUserCreateRequestDTODocumentTypeEnum]

export type ResidentialOwnerUserCreateRequestDTO = {
  /**
   * @type string | undefined
   */
  name?: string
  /**
   * @type string | undefined
   */
  rawDocumentNumber?: string
  /**
   * @type string | undefined
   */
  documentType?: ResidentialOwnerUserCreateRequestDTODocumentTypeEnum
  /**
   * @type object | undefined
   */
  account?: Account
  /**
   * @type array | undefined
   */
  units?: ResidentialUnit[]
}

File: backend/src/main/java/br/com/visitsafe/dto/user/ResidentialOwnerUserCreateRequestDTO.java
package br.com.visitsafe.dto.user;

import br.com.visitsafe.model.Account;
import br.com.visitsafe.model.enums.DocumentTypeEnum;
import br.com.visitsafe.model.unit.ResidentialUnit;
import lombok.Data;

import java.util.List;

@Data
public class ResidentialOwnerUserCreateRequestDTO {
    private String name;
    private String rawDocumentNumber;
    private DocumentTypeEnum documentType;
    private Account account;
    private List<ResidentialUnit> units;
}
